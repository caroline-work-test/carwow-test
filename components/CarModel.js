import React, {Component} from 'react';
import CarHeader from './CarHeader';

class CarModel extends Component {
    render() {
    const car = this.props.car;
    return (
<div>
      <CarHeader lead="Car details." />
      <div className="CarModel_Container">
        <div className="CarModel_Item">
          <section className="CarModel_Section">
            <div className="CarModel_Header">
              <h2>Make</h2>
              <p>{car.make}</p>
            </div>
            <div className="CarModel_Header">
              <h2>Model</h2>
              <p>{car.model}</p>
            </div>
            <h3>Rating</h3>
            <div>
              <span className="Rating_Star"/>
              <p className="Rating_Number">{car.carwow_rating}/10</p>
            </div>
            <div>
              <h3><abbr title="Recommended Retail Price">RRP</abbr></h3>
              <p className="RRP">&pound;{car.rrp}</p>
            </div>
            <button disabled className="Button">Get Offers</button>
          </section>
          <img alt={`${car.make} ${car.model}`} src={car.img_url}/>
          <section>        
            <p>{car.summary}</p>
            <h3>Available Colors</h3>
            <ul>
              {car.available_colors.map((color, idx) => {
                return (
                  <li className="Color_List" key={idx}>
                    <span className="Color_Box" style={{backgroundColor: color.colorHex}}/>
                    <p className="Color_Name">{color.colorName}</p> 
                  </li>
                )
              })}
            </ul>
            <h3>Recommended Engine</h3>
            <p>{car.recommended_engine}</p>
          </section>
        </div>
        </div>
        <style jsx >{`
        .CarModel_Section {
          display: inline-block;
        }
        .CarModel_Header {
          display: inline-block;
          margin-right: 50px;
        }
        .CarModel_Container {
          display: flex;
          flex-wrap: wrap;
          justify-content: center;
        }
        .CarModel_Item {
          display: flex;
          flex-wrap: wrap;
          flex-direction: row-reverse;
          justify-content: space-around;
          height: auto;
          width: 60%;
          margin: auto;
          padding: 10px;
          border: solid 1px #b1b1b1;
          background-color: white;
        }
        .RRP {
          font-weight: bold;
        }
        .Color_Box {
          width: 70px;
          height: 30px;
          display: block;
          border-style: solid;
          border-width: 1px;
        }
        .Rating_Star {
          background: url(../../../assets/star.svg);
          background-repeat: no-repeat;
        }
        .Color_Name {
          margin-top: 5px;
        }
        .Color_List {
          display: inline-block;
          margin-right: 30px;
        }
        .Button {
          background-color: #4CAF50;
          border: none;
          color: white;
          padding: 15px 32px;
          text-align: center;
          text-decoration: none;
          display: inline-block;
          font-size: 16px;
        }
        .Button:disabled {
          background-color: grey;
        }

        @media (max-width: 1600px) { 
          .CarModel_Item {
            width: 80%; 
          }
         }

        @media (max-width: 991.98px) { 
          .CarModel_Item {
            width: 70%;
            flex-direction: row;
            justify-content: left; 
          }
         }

        @media (max-width: 767.98px) { 
          .CarModel_Item {
            width: 80%;
            flex-direction: row;
            justify-content: left;  
          }
        }

        @media (max-width: 575.98px) {
          .CarModel_Item {
            width: 90%; 
            flex-direction: row;
            justify-content: left;         
          }
        }
      `}</style>
      </div>)}}

export default CarModel;