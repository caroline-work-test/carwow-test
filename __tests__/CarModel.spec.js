import * as React from 'react'
import {shallow, mount} from 'enzyme'
import CarModel from '../components/CarModel';
import {createSerializer} from 'enzyme-to-json';
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
//import toJson from 'enzyme-to-json';
import car from './__fixtures__/car';

describe('CarModel', () => {
  it('should render correctly', function () {
    const wrap = shallow(<CarModel car={car}/>)
    expect(wrap).toMatchSnapshot();
  })
  it('should contain button', function () {
    const wrap = shallow(<CarModel car={car}/>)
    console.log(wrap.find('button').length);
    expect(wrap.find('button')).toHaveLength(1);
  })
 
})