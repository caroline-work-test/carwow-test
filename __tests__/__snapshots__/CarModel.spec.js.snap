// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarModel should render correctly 1`] = `
<div
  className="jsx-1526905960"
>
  <CarHeader
    lead="Car details."
  />
  <div
    className="jsx-1526905960 CarModel_Container"
  >
    <div
      className="jsx-1526905960 CarModel_Item"
    >
      <section
        className="jsx-1526905960 CarModel_Section"
      >
        <div
          className="jsx-1526905960 CarModel_Header"
        >
          <h2
            className="jsx-1526905960"
          >
            Make
          </h2>
          <p
            className="jsx-1526905960"
          >
            Volvo
          </p>
        </div>
        <div
          className="jsx-1526905960 CarModel_Header"
        >
          <h2
            className="jsx-1526905960"
          >
            Model
          </h2>
          <p
            className="jsx-1526905960"
          >
            XC40
          </p>
        </div>
        <h3
          className="jsx-1526905960"
        >
          Rating
        </h3>
        <div
          className="jsx-1526905960"
        >
          <span
            className="jsx-1526905960 Rating_Star"
          />
          <p
            className="jsx-1526905960 Rating_Number"
          >
            9
            /10
          </p>
        </div>
        <div
          className="jsx-1526905960"
        >
          <h3
            className="jsx-1526905960"
          >
            <abbr
              className="jsx-1526905960"
              title="Recommended Retail Price"
            >
              RRP
            </abbr>
          </h3>
          <p
            className="jsx-1526905960 RRP"
          >
            Â£
            28965
          </p>
        </div>
        <button
          className="jsx-1526905960 Button"
          disabled={true}
        >
          Get Offers
        </button>
      </section>
      <img
        alt="Volvo XC40"
        className="jsx-1526905960"
        src="https://warm-dawn-92320.herokuapp.com/img/volvo_xc40.png"
      />
      <section
        className="jsx-1526905960"
      >
        <p
          className="jsx-1526905960"
        >
          Handsome on the outside and sophisticated on the inside, the Volvo XC40 is a spacious small SUV with smart styling and which is comfortable to drive.
        </p>
        <h3
          className="jsx-1526905960"
        >
          Available Colors
        </h3>
        <ul
          className="jsx-1526905960"
        >
          <li
            className="jsx-1526905960 Color_List"
            key="0"
          >
            <span
              className="jsx-1526905960 Color_Box"
              style={
                Object {
                  "backgroundColor": "#ffffff",
                }
              }
            />
            <p
              className="jsx-1526905960 Color_Name"
            >
              White
            </p>
          </li>
          <li
            className="jsx-1526905960 Color_List"
            key="1"
          >
            <span
              className="jsx-1526905960 Color_Box"
              style={
                Object {
                  "backgroundColor": "#d3d3d3",
                }
              }
            />
            <p
              className="jsx-1526905960 Color_Name"
            >
              Timberwolf
            </p>
          </li>
          <li
            className="jsx-1526905960 Color_List"
            key="2"
          >
            <span
              className="jsx-1526905960 Color_Box"
              style={
                Object {
                  "backgroundColor": "#cc0000",
                }
              }
            />
            <p
              className="jsx-1526905960 Color_Name"
            >
              Red Orange
            </p>
          </li>
        </ul>
        <h3
          className="jsx-1526905960"
        >
          Recommended Engine
        </h3>
        <p
          className="jsx-1526905960"
        >
          1.5 T3 163
        </p>
      </section>
    </div>
  </div>
  <JSXStyle
    id="1526905960"
  >
    .CarModel_Section.jsx-1526905960{display:inline-block;}.CarModel_Header.jsx-1526905960{display:inline-block;margin-right:50px;}.CarModel_Container.jsx-1526905960{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.CarModel_Item.jsx-1526905960{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;-webkit-box-pack:space-around;-webkit-justify-content:space-around;-ms-flex-pack:space-around;justify-content:space-around;height:auto;width:60%;margin:auto;padding:10px;border:solid 1px #b1b1b1;background-color:white;}.RRP.jsx-1526905960{font-weight:bold;}.Color_Box.jsx-1526905960{width:70px;height:30px;display:block;border-style:solid;border-width:1px;}.Rating_Star.jsx-1526905960{background:url(../../../assets/star.svg);background-repeat:no-repeat;}.Color_Name.jsx-1526905960{margin-top:5px;}.Color_List.jsx-1526905960{display:inline-block;margin-right:30px;}.Button.jsx-1526905960{background-color:#4CAF50;border:none;color:white;padding:15px 32px;text-align:center;-webkit-text-decoration:none;text-decoration:none;display:inline-block;font-size:16px;}.Button.jsx-1526905960:disabled{background-color:grey;}
  </JSXStyle>
</div>
`;
